// @progress/kendo-angular-upload/esm2020/navigation.service.mjs
/**-----------------------------------------------------------------------------------------
* Copyright © 2023 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { Injectable, EventEmitter, NgZone } from '@angular/core';
import { UploadService } from './upload.service';
import { Keys } from '@progress/kendo-angular-common';
import * as i0 from "@angular/core";
import * as i1 from "./upload.service";
/**
 * @hidden
 */
export class NavigationService {
    constructor(uploadService, zone) {
        this.uploadService = uploadService;
        this.zone = zone;
        this.onActionButtonAction = new EventEmitter();
        this.onActionButtonFocus = new EventEmitter();
        this.onFileAction = new EventEmitter();
        this.onFileFocus = new EventEmitter();
        this.onTab = new EventEmitter();
        this.onWrapperFocus = new EventEmitter();
        this.onSelectButtonFocus = new EventEmitter();
        this.actionButtonsVisible = false;
        this.focused = false;
        this._focusedIndex = -1;
    }
    action(event) {
        const key = event.keyCode;
        return this.keyBindings[key];
    }
    process(event) {
        const handler = this.action(event);
        if (handler) {
            handler(event);
        }
    }
    computeKeys(direction) {
        this.keyBindings = {
            [Keys.Space]: () => this.handleSpace(),
            [Keys.Enter]: (event) => this.handleEnter(event),
            [Keys.Escape]: () => this.handleEscape(),
            [Keys.Delete]: () => this.handleDelete(),
            [Keys.Tab]: (event) => this.handleTab(event.shiftKey),
            [Keys.ArrowUp]: (event) => this.handleUp(event),
            [Keys.ArrowDown]: (event) => this.handleDown(event),
            [this.invertKeys(direction, Keys.ArrowLeft, Keys.ArrowRight)]: () => this.handleLeft(),
            [this.invertKeys(direction, Keys.ArrowRight, Keys.ArrowLeft)]: () => this.handleRight()
        };
    }
    invertKeys(direction, original, inverted) {
        return direction === 'rtl' ? inverted : original;
    }
    focusSelectButton() {
        this.focused = true;
        this._focusedIndex = -1;
        this.onSelectButtonFocus.emit();
    }
    handleEnter(event) {
        if (this.lastIndex >= 0) {
            this.zone.run(() => {
                if (this.focusedIndex <= this.lastFileIndex) {
                    this.onFileAction.emit(Keys.Enter);
                    return;
                }
                if (this.actionButtonsVisible && this.focusedIndex <= this.lastIndex) {
                    event.preventDefault();
                    this.onActionButtonAction.emit(this.focusedIndex < this.lastIndex ? "clear" : "upload");
                }
            });
        }
    }
    handleSpace() {
        if (this.focusedIndex >= 0 && this.focusedIndex <= this.lastFileIndex) {
            this.zone.run(() => this.onFileAction.emit(Keys.Space));
        }
    }
    handleDelete() {
        if (this.focusedIndex >= 0 && this.focusedIndex <= this.lastFileIndex) {
            this.zone.run(() => {
                this.onFileAction.emit(Keys.Delete);
            });
        }
    }
    handleEscape() {
        if (this.focusedIndex >= 0 && this.focusedIndex <= this.lastFileIndex) {
            this.onFileAction.emit(Keys.Escape);
        }
    }
    handleLeft() {
        if (this.actionButtonsVisible && this.focusedIndex === this.lastIndex) {
            this.focusedIndex -= 1;
            this.zone.run(() => {
                this.onActionButtonFocus.emit("clear");
            });
        }
    }
    handleRight() {
        if (this.actionButtonsVisible && this.focusedIndex === this.lastIndex - 1) {
            this.focusedIndex += 1;
            this.zone.run(() => {
                this.onActionButtonFocus.emit("upload");
            });
        }
    }
    handleTab(shifted) {
        if (this.focusedIndex === -1) {
            this.focusedIndex = this.lastFileIndex + 1;
            this.zone.run(() => this.onActionButtonFocus.emit("clear"));
            return;
        }
        if (this.focusedIndex === this.lastFileIndex + 1) {
            this.focusedIndex += 1;
            this.zone.run(() => this.onActionButtonFocus.emit("upload"));
            return;
        }
        if (this.focusedIndex === this.lastIndex && shifted) {
            this.focusedIndex -= 1;
            return;
        }
        if (this.focusedIndex === this.lastFileIndex + 1 && shifted) {
            this.focusedIndex = -1;
            this.zone.run(() => this.onSelectButtonFocus.emit());
        }
    }
    handleDown(event) {
        if (this.lastIndex >= 0 && this.focusedIndex < this.lastIndex) {
            event.preventDefault();
            this.zone.run(() => {
                if (this.focusedIndex < this.lastFileIndex) {
                    this.focusedIndex += 1;
                    this.onFileFocus.emit(this.focusedIndex);
                    return;
                }
                if (this.actionButtonsVisible && this.focusedIndex === this.lastFileIndex) {
                    this.focusedIndex += 1;
                    this.onActionButtonFocus.emit("clear");
                }
            });
        }
    }
    handleUp(event) {
        if (this.lastIndex >= 0 && this.focusedIndex > -1) {
            event.preventDefault();
            this.zone.run(() => {
                this.focusedIndex -= 1;
                if (this.focusedIndex === -1) {
                    this.onSelectButtonFocus.emit();
                    return;
                }
                if (this.focusedIndex <= this.lastFileIndex) {
                    this.onFileFocus.emit(this.focusedIndex);
                    return;
                }
                if (this.actionButtonsVisible && this.focusedIndex <= this.lastIndex) {
                    this.focusedIndex = this.lastFileIndex;
                    this.onFileFocus.emit(this.focusedIndex);
                }
            });
        }
    }
    get focusedIndex() {
        return this._focusedIndex;
    }
    set focusedIndex(index) {
        if (!this.focused) {
            this.onWrapperFocus.emit();
        }
        this._focusedIndex = index;
        this.focused = true;
        if (this._focusedIndex >= 0 && this._focusedIndex <= this.lastFileIndex) {
            this.onFileFocus.emit(index);
        }
    }
    get lastFileIndex() {
        return this.actionButtonsVisible ? this.lastIndex - 2 : this.lastIndex;
    }
    get lastIndex() {
        const fileCount = this.uploadService.files.count;
        return this.actionButtonsVisible ? fileCount + 1 : fileCount - 1;
    }
}
NavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: NavigationService, deps: [{ token: i1.UploadService }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Injectable });
NavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: NavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: NavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.UploadService }, { type: i0.NgZone }]; } });
