// @progress/kendo-angular-editor/esm2020/dialogs/file-link-dialog.component.mjs
/**-----------------------------------------------------------------------------------------
* Copyright © 2023 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { Component, Input, ViewChild } from '@angular/core';
import { FormGroup, FormControl, Validators } from '@angular/forms';
import { DialogContentBase, DialogRef } from '@progress/kendo-angular-dialog';
import { getMark, getSelectionText, expandSelection } from '@progress/kendo-editor-common';
import { isPresent } from '../util';
import { EditorLocalizationService } from '../localization/editor-localization.service';
import { TextBoxComponent } from '@progress/kendo-angular-inputs';
import * as i0 from "@angular/core";
import * as i1 from "@progress/kendo-angular-dialog";
import * as i2 from "../localization/editor-localization.service";
import * as i3 from "@progress/kendo-angular-label";
import * as i4 from "@progress/kendo-angular-inputs";
import * as i5 from "@progress/kendo-angular-buttons";
import * as i6 from "@angular/forms";
import * as i7 from "@angular/common";
/**
 * @hidden
 */
export class FileLinkDialogComponent extends DialogContentBase {
    constructor(dialog, localization) {
        super(dialog);
        this.dialog = dialog;
        this.localization = localization;
        this.linkForm = new FormGroup({
            'href': new FormControl('', Validators.required),
            'text': new FormControl({ value: '', disabled: true }, Validators.required),
            'title': new FormControl('')
        });
    }
    ngAfterViewInit() {
        Promise.resolve(null).then(() => {
            this.hrefInput.focus();
        });
    }
    onCancelAction() {
        this.dialog.close();
    }
    onConfirmAction() {
        const linkData = this.getData();
        this.editor.exec(this.command, linkData);
        this.dialog.close();
        this.editor.view.focus();
    }
    get titleText() {
        return this.localization.get(this.command);
    }
    setData(state, options) {
        if (this.command === 'createLink') {
            this.linkForm.addControl('target', new FormControl());
        }
        const linkMark = getMark(state, state.schema.marks['link']);
        if (linkMark) {
            // const linkMarkRange = getMarkRange(state.selection.$cursor, schema.marks.link);
            // const mark = parentNode.child(cursorNodeIndex).marks.find(m => m.type === markType);
            this.linkForm.reset({
                href: linkMark.attrs['href'],
                title: linkMark.attrs['title'],
                target: isPresent(linkMark.attrs['target']),
                text: this.setLinkText(state)
            });
            return;
        }
        if (state.selection.empty) {
            const currentState = options.applyToWord ? expandSelection(state, () => { }, options).state : state;
            if (!currentState.selection.empty) {
                this.linkForm.patchValue({
                    'text': getSelectionText(currentState)
                });
            }
        }
        else {
            this.linkForm.patchValue({
                'text': getSelectionText(state)
            });
        }
    }
    textForWithPrefix(key) {
        const prefix = this.command === 'createLink' ? 'link' : 'file';
        return this.textFor(prefix + key);
    }
    textFor(key) {
        return this.localization.get(key);
    }
    setLinkText(state) {
        const selection = state.selection;
        if (selection.empty && selection.$cursor) {
            const cursor = selection.$cursor;
            const cursorNodeIndex = cursor.index();
            const parentNode = cursor.parent;
            return parentNode.child(cursorNodeIndex).text;
        }
        else {
            return getSelectionText(state);
        }
    }
    getData() {
        const linkData = this.linkForm.value;
        if (isPresent(this.linkForm.controls['target'])) {
            linkData.target = linkData.target ? '_blank' : null;
        }
        return linkData;
    }
}
FileLinkDialogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: FileLinkDialogComponent, deps: [{ token: i1.DialogRef }, { token: i2.EditorLocalizationService }], target: i0.ɵɵFactoryTarget.Component });
FileLinkDialogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.12", type: FileLinkDialogComponent, selector: "ng-component", inputs: { editor: "editor", command: "command" }, viewQueries: [{ propertyName: "hrefInput", first: true, predicate: ["hrefInput"], descendants: true }], usesInheritance: true, ngImport: i0, template: `
        <kendo-dialog-titlebar (close)="onCancelAction()">
            {{ titleText }}
        </kendo-dialog-titlebar>
        <form class="k-form k-form-md k-form-horizontal" novalidate [formGroup]="linkForm">
            <div class="k-form-field">
                <div class="k-form-label">
                    <kendo-label [for]="hrefInput" [text]="textForWithPrefix('WebAddress')"></kendo-label>
                </div>
                <div class="k-form-field-wrap">
                    <kendo-textbox #hrefInput formControlName="href"></kendo-textbox>
                </div>
            </div>
            <div class="k-form-field">
                <div class="k-form-label">
                    <kendo-label [for]="textInput" [text]="textForWithPrefix('Text')"></kendo-label>
                </div>
                <div class="k-form-field-wrap">
                    <kendo-textbox #textInput formControlName="text"></kendo-textbox>
                </div>
            </div>
            <div class="k-form-field">
                <div class="k-form-label">
                    <kendo-label [for]="titleInput" [text]="textForWithPrefix('Title')"></kendo-label>
                </div>
                <div class="k-form-field-wrap">
                    <kendo-textbox #titleInput formControlName="title"></kendo-textbox>
                </div>
            </div>
            <ng-container *ngIf="command === 'createLink'">
                <div class="k-form-field">
                    <div class="k-form-label"></div>
                    <div class="k-form-field-wrap">
                        <input type="checkbox" id="k-target-blank" kendoCheckBox formControlName="target" />
                        <label class="k-checkbox-label" for="k-target-blank" [style.display]="'inline'">{{ textForWithPrefix('OpenInNewWindow') }}</label>
                    </div>
                </div>
            </ng-container>
        </form>
        <kendo-dialog-actions class="k-actions-end k-actions-horizontal k-window-actions" [layout]="'start'">
            <button kendoButton (click)="onCancelAction()">{{ textFor('dialogCancel') }}</button>
            <button kendoButton [disabled]="linkForm.invalid" (click)="onConfirmAction()" [primary]="true">
                {{ textFor('dialogInsert') }}
            </button>
        </kendo-dialog-actions>
    `, isInline: true, components: [{ type: i1.DialogTitleBarComponent, selector: "kendo-dialog-titlebar", inputs: ["id", "closeTitle"], outputs: ["close"] }, { type: i3.LabelComponent, selector: "kendo-label", inputs: ["text", "for", "optional"], exportAs: ["kendoLabel"] }, { type: i4.TextBoxComponent, selector: "kendo-textbox", inputs: ["focusableId", "title", "disabled", "readonly", "tabindex", "value", "selectOnFocus", "showSuccessIcon", "showErrorIcon", "clearButton", "successIcon", "errorIcon", "clearButtonIcon", "size", "rounded", "fillMode", "tabIndex", "placeholder", "maxlength"], outputs: ["valueChange", "inputFocus", "inputBlur", "focus", "blur"], exportAs: ["kendoTextBox"] }, { type: i1.DialogActionsComponent, selector: "kendo-dialog-actions", inputs: ["actions", "layout"], outputs: ["action"] }, { type: i5.Button, selector: "button[kendoButton], span[kendoButton], kendo-button", inputs: ["toggleable", "togglable", "selected", "tabIndex", "imageUrl", "iconClass", "icon", "disabled", "size", "rounded", "fillMode", "themeColor", "svgIcon", "role", "primary", "look"], outputs: ["selectedChange", "click"], exportAs: ["kendoButton"] }], directives: [{ type: i6.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i6.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i6.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i6.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i6.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { type: i7.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i6.CheckboxControlValueAccessor, selector: "input[type=checkbox][formControlName],input[type=checkbox][formControl],input[type=checkbox][ngModel]" }, { type: i4.CheckBoxDirective, selector: "input[kendoCheckBox]", inputs: ["size", "rounded"] }, { type: i3.LabelDirective, selector: "label[for]", inputs: ["for"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: FileLinkDialogComponent, decorators: [{
            type: Component,
            args: [{
                    template: `
        <kendo-dialog-titlebar (close)="onCancelAction()">
            {{ titleText }}
        </kendo-dialog-titlebar>
        <form class="k-form k-form-md k-form-horizontal" novalidate [formGroup]="linkForm">
            <div class="k-form-field">
                <div class="k-form-label">
                    <kendo-label [for]="hrefInput" [text]="textForWithPrefix('WebAddress')"></kendo-label>
                </div>
                <div class="k-form-field-wrap">
                    <kendo-textbox #hrefInput formControlName="href"></kendo-textbox>
                </div>
            </div>
            <div class="k-form-field">
                <div class="k-form-label">
                    <kendo-label [for]="textInput" [text]="textForWithPrefix('Text')"></kendo-label>
                </div>
                <div class="k-form-field-wrap">
                    <kendo-textbox #textInput formControlName="text"></kendo-textbox>
                </div>
            </div>
            <div class="k-form-field">
                <div class="k-form-label">
                    <kendo-label [for]="titleInput" [text]="textForWithPrefix('Title')"></kendo-label>
                </div>
                <div class="k-form-field-wrap">
                    <kendo-textbox #titleInput formControlName="title"></kendo-textbox>
                </div>
            </div>
            <ng-container *ngIf="command === 'createLink'">
                <div class="k-form-field">
                    <div class="k-form-label"></div>
                    <div class="k-form-field-wrap">
                        <input type="checkbox" id="k-target-blank" kendoCheckBox formControlName="target" />
                        <label class="k-checkbox-label" for="k-target-blank" [style.display]="'inline'">{{ textForWithPrefix('OpenInNewWindow') }}</label>
                    </div>
                </div>
            </ng-container>
        </form>
        <kendo-dialog-actions class="k-actions-end k-actions-horizontal k-window-actions" [layout]="'start'">
            <button kendoButton (click)="onCancelAction()">{{ textFor('dialogCancel') }}</button>
            <button kendoButton [disabled]="linkForm.invalid" (click)="onConfirmAction()" [primary]="true">
                {{ textFor('dialogInsert') }}
            </button>
        </kendo-dialog-actions>
    `
                }]
        }], ctorParameters: function () { return [{ type: i1.DialogRef }, { type: i2.EditorLocalizationService }]; }, propDecorators: { editor: [{
                type: Input
            }], command: [{
                type: Input
            }], hrefInput: [{
                type: ViewChild,
                args: ['hrefInput']
            }] } });
