// @progress/kendo-angular-charts/esm2020/sparkline.module.mjs
/**-----------------------------------------------------------------------------------------
* Copyright © 2023 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { NgModule } from '@angular/core';
import { ChartModule } from './chart.module';
import { SPARKLINE_DIRECTIVES } from './sparkline.directives';
import { CommonModule } from '@angular/common';
import { PopupModule } from '@progress/kendo-angular-popup';
import { ResizeSensorModule } from '@progress/kendo-angular-common';
import * as i0 from "@angular/core";
import * as i1 from "./sparkline.component";
/**
 * A [module](link:site.data.urls.angular['ngmoduleapi']) that includes the Sparkline component and directives.
 *
 * Imports the SparklineModule into your application
 * [root module](link:site.data.urls.angular['ngmodules']#angular-modularity) or any other sub-module
 * that will use the Sparkline component.
 *
 * @example
 * ```ts-no-run
 * import { NgModule } from '@angular/core';
 * import { BrowserModule } from '@angular/platform-browser';
 * import { SparklineModule } from '@progress/kendo-angular-charts';
 * import { AppComponent } from './app.component';
 *
 * _@NgModule({
 *     bootstrap:    [AppComponent],
 *     declarations: [AppComponent],
 *     imports:      [BrowserModule, SparklineModule]
 * })
 * export class AppModule {
 * }
 * ```
 */
export class SparklineModule {
}
SparklineModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: SparklineModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SparklineModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: SparklineModule, declarations: [i1.SparklineComponent], imports: [ChartModule, CommonModule, PopupModule, ResizeSensorModule], exports: [i1.SparklineComponent, ChartModule] });
SparklineModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: SparklineModule, imports: [[ChartModule, CommonModule, PopupModule, ResizeSensorModule], ChartModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: SparklineModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SPARKLINE_DIRECTIVES],
                    exports: [SPARKLINE_DIRECTIVES, ChartModule],
                    imports: [ChartModule, CommonModule, PopupModule, ResizeSensorModule]
                }]
        }] });
