// @progress/kendo-angular-charts/esm2020/stock-chart.module.mjs
/**-----------------------------------------------------------------------------------------
* Copyright © 2023 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { NgModule } from '@angular/core';
import { ChartModule } from './chart.module';
import { CommonModule } from '@angular/common';
import { PopupModule } from '@progress/kendo-angular-popup';
import { ResizeSensorModule } from '@progress/kendo-angular-common';
import { STOCK_CHART_DIRECTIVES } from './stock-chart.directives';
import * as i0 from "@angular/core";
import * as i1 from "./stock-chart.component";
import * as i2 from "./stock-chart/navigator.component";
import * as i3 from "./stock-chart/navigator/category-axis.component";
import * as i4 from "./stock-chart/navigator/category-axis/crosshair.component";
import * as i5 from "./stock-chart/navigator/category-axis/crosshair.tooltip.component";
import * as i6 from "./stock-chart/navigator/category-axis/labels.component";
import * as i7 from "./stock-chart/navigator/category-axis/notes.component";
import * as i8 from "./stock-chart/navigator/category-axis/notes.icon.component";
import * as i9 from "./stock-chart/navigator/category-axis/notes.label.component";
import * as i10 from "./stock-chart/navigator/category-axis/select.component";
import * as i11 from "./stock-chart/navigator/category-axis/title.component";
import * as i12 from "./stock-chart/navigator/hint.component";
import * as i13 from "./stock-chart/navigator/pane.component";
import * as i14 from "./stock-chart/navigator/pane/title.component";
import * as i15 from "./stock-chart/navigator/select.component";
import * as i16 from "./stock-chart/navigator/series.component";
import * as i17 from "./stock-chart/navigator/series-item.component";
import * as i18 from "./stock-chart/navigator/series-item/error-bars.component";
import * as i19 from "./stock-chart/navigator/series-item/extremes.component";
import * as i20 from "./stock-chart/navigator/series-item/highlight.component";
import * as i21 from "./stock-chart/navigator/series-item/labels.component";
import * as i22 from "./stock-chart/navigator/series-item/labels.from.component";
import * as i23 from "./stock-chart/navigator/series-item/labels.to.component";
import * as i24 from "./stock-chart/navigator/series-item/markers.component";
import * as i25 from "./stock-chart/navigator/series-item/notes.component";
import * as i26 from "./stock-chart/navigator/series-item/notes.icon.component";
import * as i27 from "./stock-chart/navigator/series-item/notes.label.component";
import * as i28 from "./stock-chart/navigator/series-item/outliers.component";
import * as i29 from "./stock-chart/navigator/series-item/tooltip.component";
/**
 * A [module](link:site.data.urls.angular['ngmoduleapi']) that includes the StockChart component and directives.
 *
 * Imports the StockChartModule into your application
 * [root module](link:site.data.urls.angular['ngmodules']#angular-modularity) or any other sub-module
 * that will use the StockChart component.
 *
 * @example
 * ```ts-no-run
 * import { NgModule } from '@angular/core';
 * import { BrowserModule } from '@angular/platform-browser';
 * import { StockChartModule } from '@progress/kendo-angular-charts';
 * import { AppComponent } from './app.component';
 *
 * _@NgModule({
 *     bootstrap:    [AppComponent],
 *     declarations: [AppComponent],
 *     imports:      [BrowserModule, StockChartModule]
 * })
 * export class AppModule {
 * }
 * ```
 */
export class StockChartModule {
}
StockChartModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: StockChartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StockChartModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: StockChartModule, declarations: [i1.StockChartComponent, i2.NavigatorComponent, i3.NavigatorCategoryAxisComponent, i4.NavigatorCategoryAxisCrosshairComponent, i5.NavigatorCategoryAxisCrosshairTooltipComponent, i6.NavigatorCategoryAxisLabelsComponent, i7.NavigatorCategoryAxisNotesComponent, i8.NavigatorCategoryAxisNotesIconComponent, i9.NavigatorCategoryAxisNotesLabelComponent, i10.NavigatorCategoryAxisSelectComponent, i11.NavigatorCategoryAxisTitleComponent, i12.NavigatorHintComponent, i13.NavigatorPaneComponent, i14.NavigatorPaneTitleComponent, i15.NavigatorSelectComponent, i16.NavigatorSeriesComponent, i17.NavigatorSeriesItemComponent, i18.NavigatorSeriesErrorBarsComponent, i19.NavigatorSeriesExtremesComponent, i20.NavigatorSeriesHighlightComponent, i21.NavigatorSeriesLabelsComponent, i22.NavigatorSeriesLabelsFromComponent, i23.NavigatorSeriesLabelsToComponent, i24.NavigatorSeriesMarkersComponent, i25.NavigatorSeriesNotesComponent, i26.NavigatorSeriesNotesIconComponent, i27.NavigatorSeriesNotesLabelComponent, i28.NavigatorSeriesOutliersComponent, i29.NavigatorSeriesTooltipComponent], imports: [ChartModule, CommonModule, PopupModule, ResizeSensorModule], exports: [i1.StockChartComponent, i2.NavigatorComponent, i3.NavigatorCategoryAxisComponent, i4.NavigatorCategoryAxisCrosshairComponent, i5.NavigatorCategoryAxisCrosshairTooltipComponent, i6.NavigatorCategoryAxisLabelsComponent, i7.NavigatorCategoryAxisNotesComponent, i8.NavigatorCategoryAxisNotesIconComponent, i9.NavigatorCategoryAxisNotesLabelComponent, i10.NavigatorCategoryAxisSelectComponent, i11.NavigatorCategoryAxisTitleComponent, i12.NavigatorHintComponent, i13.NavigatorPaneComponent, i14.NavigatorPaneTitleComponent, i15.NavigatorSelectComponent, i16.NavigatorSeriesComponent, i17.NavigatorSeriesItemComponent, i18.NavigatorSeriesErrorBarsComponent, i19.NavigatorSeriesExtremesComponent, i20.NavigatorSeriesHighlightComponent, i21.NavigatorSeriesLabelsComponent, i22.NavigatorSeriesLabelsFromComponent, i23.NavigatorSeriesLabelsToComponent, i24.NavigatorSeriesMarkersComponent, i25.NavigatorSeriesNotesComponent, i26.NavigatorSeriesNotesIconComponent, i27.NavigatorSeriesNotesLabelComponent, i28.NavigatorSeriesOutliersComponent, i29.NavigatorSeriesTooltipComponent, ChartModule] });
StockChartModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: StockChartModule, imports: [[ChartModule, CommonModule, PopupModule, ResizeSensorModule], ChartModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: StockChartModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [STOCK_CHART_DIRECTIVES],
                    exports: [STOCK_CHART_DIRECTIVES, ChartModule],
                    imports: [ChartModule, CommonModule, PopupModule, ResizeSensorModule]
                }]
        }] });
